services-discovery/src/main/java/io/scalecube/services/discovery/ScalecubeServiceDiscovery.java

```diff
@@ -86,8 +86,10 @@ public ServiceEndpoint endpoint() {
 
   @Override
   public Mono<Void> shutdown() {
-    sink.complete();
-    return Mono.fromFuture(cluster.shutdown());
+    return Mono.defer(() -> {
+      sink.complete();
+      return cluster != null ? Mono.fromFuture(cluster.shutdown()) : Mono.empty();
```

Comment:
In accordance with similar changes maybe better to use `Optional`?

Reply:
@dmytro-lazebnyi yep